packages/web-components/src/slider/slider.ts

```diff
@@ -324,80 +601,72 @@ export class Slider extends FormAssociatedSlider implements SliderConfiguration
    */
   public decrement(): void {
     const newVal =
-      this.direction !== Direction.rtl && this.orientation !== Orientation.vertical
-        ? Number(this.value) - Number(this.stepValue)
-        : Number(this.value) - Number(this.stepValue);
+      this.direction !== Direction.rtl
+        ? Number(this.value) - Number(this.stepAsNumber)
+        : Number(this.value) + Number(this.stepAsNumber);
     const decrementedVal: number = this.convertToConstrainedValue(newVal);
-    const decrementedValString: string = decrementedVal > Number(this.min) ? `${decrementedVal}` : `${this.min}`;
+    const decrementedValString: string =
+      decrementedVal > this.minAsNumber ? `${decrementedVal}` : `${this.minAsNumber}`;
     this.value = decrementedValString;
   }
 
-  public keypressHandler = (e: KeyboardEvent): void => {
-    if (this.readOnly || this.disabled) {
+  public keypressHandler = (event: KeyboardEvent): void => {
+    if (this.disabled) {
       return;
     }
 
-    if (e.key === keyHome) {
-      e.preventDefault();
-      this.direction !== Direction.rtl && this.orientation !== Orientation.vertical
-        ? (this.value = `${this.min}`)
-        : (this.value = `${this.max}`);
-    } else if (e.key === keyEnd) {
-      e.preventDefault();
-      this.direction !== Direction.rtl && this.orientation !== Orientation.vertical
-        ? (this.value = `${this.max}`)
-        : (this.value = `${this.min}`);
-    } else if (!e.shiftKey) {
-      switch (e.key) {
+    if (event.key === keyHome) {
+      event.preventDefault();
+      this.value =
+        this.direction !== Direction.rtl && this.orientation !== Orientation.vertical
+          ? `${this.minAsNumber}`
+          : `${this.maxAsNumber}`;
+    } else if (event.key === keyEnd) {
+      event.preventDefault();
+      this.value =
+        this.direction !== Direction.rtl && this.orientation !== Orientation.vertical
+          ? `${this.maxAsNumber}`
+          : `${this.minAsNumber}`;
+    } else if (!event.shiftKey) {
+      switch (event.key) {
         case keyArrowRight:
         case keyArrowUp:
-          e.preventDefault();
+          event.preventDefault();
           this.increment();
           break;
         case keyArrowLeft:
         case keyArrowDown:
-          e.preventDefault();
+          event.preventDefault();
           this.decrement();
           break;
       }
     }
   };
 
-  /**
-   * Gets the actual step value for the slider
-   *
-   */
-  private get stepValue(): number {
-    return this.step === undefined ? 1 : this.step;
-  }
-
   /**
    * Places the thumb based on the current value
    *
    * @public
```

Comment:
Should this be marked as `@internal`?

Reply:
Ah since it's a private method, I just removed the `@public` tag, assuming all private properties/methods don't need explicit `@internal` tags.